dask_searchcv/adaptive.py

```diff
@@ -0,0 +1,385 @@
+        r_i = r * eta**i
+        iters += r_i
+        if verbose:
+            msg = ('Training {n} models for {r} iterations during iteration {i} '
+                   'of bracket={s}')
+            print(msg.format(n=n_i, r=r_i, i=i, s=s))
+
+        futures = {k: client.submit(_train, model, *data['train'], *data['val'],
+                                    max_iter=r_i, classes=classes, s=s, i=i,
+                                    k=k, verbose=verbose, dry_run=dry_run,
+                                    **fit_kwargs)
+                      for k, model in models.items()}
+        results = client.gather(futures)
+
+        val_scores = {k: r[0] for k, r in results.items()}
+        times += [{'id': k, **r[1]} for k, r in results.items()]
+
+        final_val_scores.update(val_scores)
+        history += [{'s': s, 'i': i, 'val_score': val_scores[k], 'model_id': k,
+                     'iters': iters, 'num_models': len(val_scores), **params[k]}
+                     for k, model, in models.items()]
+
+        models = _top_k(models, val_scores, k=max(1, math.floor(n_i / eta)))
+
+        best_ = {k: v.get() for k, v in best.items()}
+        if max(final_val_scores.values()) > best_['val_score']:
+            max_loss_key = max(final_val_scores, key=final_val_scores.get)
+            best_['val_score'] = final_val_scores[max_loss_key]
+            best_['config'] = params[max_loss_key]
+            [best[k].set(v) for k, v in best_.items()]
+            if verbose:
+                print("New best val_score={} found".format(best_['val_score']))
+
+        if len(models) == 1:
+            break
+
+    times = toolz.groupby('id', times)
+    times = {k: [toolz.keyfilter(lambda key: key != 'id', vi) for vi in v]
+             for k, v in times.items()}
+    times = {k: toolz.merge_with(sum, v) for k, v in times.items()}
+    return {'history': history, 'final_val_scores': final_val_scores,
+            'params': params, 'models': models, 'times': times}
+
+
+class Hyperband(DaskBaseSearchCV):
+    def __init__(self, model, params, max_iter=81, eta=3,
+                 run_in_parallel=True, **kwargs):
+        """
+        An adaptive model selection method that trains models with
+        ``model.partial_fit`` and evaluates models with ``model.score``.
+
+        Parameters
+        ----------
+        params :
+            Search space of various parameters. This mirrors the sklearn API.
+
+        model :
+            The base model to try.
+
+        max_iter : int
+            The maximum number of iterations to train a model on, as determined
+            by ``model.partial_fit``.
+
+        X, y : np.ndarray, np.ndarray
+            Train data, features ``X`` and labels ``y``.
+
+        eta : optional, int
+            How aggressive to be while search. The default is 3, and theory
+            suggests that ``eta=e=2.718...``. Changing this is not recommended.
+
+        run_in_parallel : bool, optional
```

Comment:
Maybe call this `n_jobs` and have the default be `-1`. Then `run_in_parallel = n_jobs == -1`.

Reply:
I'm seeing some support for `n_jobs` in the distributed Client, but I'll leave that for a future PR. Right now `n_jobs` has to be either 0 or -1.